src/core/openthread.cpp

```diff
@@ -1256,9 +1256,17 @@ void otSetReceiveIp6DatagramFilterEnabled(otInstance *aInstance, bool aEnabled)
 ThreadError otSendIp6Datagram(otInstance *aInstance, otMessage aMessage)
 {
     ThreadError error;
+    Message *message;
+    Ip6::Header header;
 
     otLogFuncEntry();
 
+    message = static_cast<Message *>(aMessage);
+    assert(message->GetLength() >= sizeof(header));
+
+    message->Read(0, sizeof(header), &header);
+    assert(!header.GetDestination().IsRealmLocalMulticast() || header.GetNextHeader() != Ip6::kProtoHopOpts);
```

Comment:
@nibanks, what you say is true if we assume that the source always puts the MPL Option at the end.  However, there is currently no specification that requires the MPL Option to be at the end and an independent Thread stack may put the MPL Option somewhere else.  What should OpenThread pass up to the host in this case?  The only acceptable way I can think of is by replacing the MPL Option with a PadN Option.